### ISSUE TITLE
Diagnostic for using non-POD type with va_start needs improvement

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#6857]
|  |  |
| --- | --- |
| Bugzilla Link | [6485](https://llvm.org/bz6485) |
| Version | trunk |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |
| CC | @DougGregor,@zmodem |

## Extended Description 
coppro@overthere:~$ clang -x c++ -
```c++
#include <stdarg.h>               

struct foo { virtual void bar(); };

void baz (foo f, ...) {
  va_list v;
  va_start(v, f);
}
```
```
<stdin>:7:3: error: cannot pass object of non-POD type 'struct foo' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
  va_start(v, f);
  ^
<stdin>:7:15: note: instantiated from:
  va_start(v, f);
              ^
1 diagnostic generated.
```

GCC does not issue a diagnostic here, however, it is good that clang do so ([support.runtime]/3). The diagnostic could bear improvement, however - prehaps to something along the lines of "cannot use va_start on a parameter of non-POD type 'struct foo'".

**Comment by philnik777** [comment]
Clang accepts this code since 2.8 (https://godbolt.org/z/qYrr6c46d), so I guess this isn't relevant anymore.
